parsers/openapi.c

```diff
@@ -0,0 +1,363 @@
+							 yaml_token_t *token)
+{
+	while (openapi->type_stack)
+		popBlockType (openapi, token);
+}
+
+static bool stateStackMatch (struct yamlBlockTypeStack *stack,
+							 const enum openapiKeys *expectation,
+							 unsigned int length
+							 )
+{
+	if (length == 0)
+	{
+		if (stack == NULL)
+			return true;
+		else
+			return false;
+	}
+
+	if (stack == NULL)
+		return false;
+
+	if (stack->key == expectation[0])
+		return stateStackMatch (stack->next, expectation + 1, length - 1);
+	else
+		return false;
+}
+
+static bool scalarNeq (yaml_token_t *token, unsigned int len, const char* val)
+{
+	if ((token->data.scalar.length == len)
+		&& (strncmp (val, (char *)token->data.scalar.value, len) == 0))
+		return true;
+	else
+		return false;
+}
+
+static enum openapiKeys parseKey(yaml_token_t *token)
+{
+  if (scalarNeq(token, 10, "components"))
+	  return KEY_COMPONENTS;
+  else if (scalarNeq(token, 7, "schemas"))
+	  return KEY_SCHEMAS;
+  else if (scalarNeq(token, 5, "paths"))
+	  return KEY_PATHS;
+  else if (scalarNeq(token, 11, "definitions"))
+	  return KEY_DEFINITIONS;
+  else if (scalarNeq(token, 10, "parameters"))
+	  return KEY_PARAMETERS;
+  else if (scalarNeq(token, 9, "responses"))
+	  return KEY_RESPONSES;
+  else
+	  return KEY_UNKNOWN;
+}
+
+static void printStack(struct yamlBlockTypeStack* stack)
+{
+  if (!stack) {
+	  puts("");
+	  return;
+  }
+
+  printf("[%d] - ", stack->key);
+  printStack(stack->next);
+}
+
+
+struct tagSource {
+	openapiKind kind;
+	const enum openapiKeys* keys;
+	size_t countKeys;
+};
+
+static const enum openapiKeys pathKeys[] = {
+	KEY_UNKNOWN,
+	KEY_PATHS,
+};
+
+static const enum openapiKeys responses3Keys[] = {
+	KEY_UNKNOWN,
+	KEY_RESPONSES,
+	KEY_COMPONENTS,
+};
+
+static const enum openapiKeys responses2Keys[] = {
+	KEY_UNKNOWN,
+	KEY_RESPONSES,
+};
+
+static const enum openapiKeys parameters3Keys[] = {
+	KEY_UNKNOWN,
+	KEY_PARAMETERS,
+	KEY_COMPONENTS,
+};
+
+static const enum openapiKeys parameters2Keys[] = {
+	KEY_UNKNOWN,
+	KEY_PARAMETERS,
+};
+
+static const enum openapiKeys schemas3Keys[] = {
+	KEY_UNKNOWN,
+	KEY_SCHEMAS,
+	KEY_COMPONENTS,
+};
+
+static const enum openapiKeys definitions2Keys[] = {
+	KEY_UNKNOWN,
+	KEY_DEFINITIONS,
+};
+
+const struct tagSource tagSources[] = {
+	{
+		KIND_PATH,
+		pathKeys,
+		ARRAY_SIZE (pathKeys),
+	},
+	{
+		KIND_RESPONSE,
+		responses3Keys,
+		ARRAY_SIZE (responses3Keys),
+	},
+	{
+		KIND_RESPONSE,
+		responses2Keys,
+		ARRAY_SIZE (responses2Keys),
+	},
+	{
+		KIND_PARAMETER,
+		parameters3Keys,
+		ARRAY_SIZE (parameters3Keys),
+	},
+	{
+		KIND_PARAMETER,
+		parameters2Keys,
+		ARRAY_SIZE (parameters2Keys),
+	},
+	{
+		KIND_SCHEMA,
+		schemas3Keys,
+		ARRAY_SIZE (schemas3Keys),
+	},
+	{
+		KIND_SCHEMA,
+		definitions2Keys,
+		ARRAY_SIZE (definitions2Keys),
+	}
+};
+
+static void handleKey(struct sOpenApiSubparser *openapi,
+											 yaml_token_t *token)
+{
+	int i;
+	for (i = 0; i < ARRAY_SIZE(tagSources); i++) {
```

Comment:
Will you maintain this parser？
If the answer is "yes", you can choose your favorite C coding style.
If the answer is "no", I would like you to follow the ctags' coding style like:
```
   for (i = 0; i < ARRAY_SIZE(tagSources); i++)
   {
      ...
```
See the position of `{`.

Reply:
fixed